Does not compile on a Mac due to unix.SIOCSARP dependency

```sh
$ go install ./...
go: downloading github.com/rs/zerolog v1.23.0
go: downloading github.com/Masterminds/sprig v2.22.0+incompatible
go: downloading github.com/digitalrebar/tftp v0.0.0-20200914190809-39d58dc90c67
go: downloading github.com/coreos/go-systemd v0.0.0-20191104093116-d3cd4ed1dbcf
go: downloading github.com/insomniacslk/dhcp v0.0.0-20210621130208-1cac67f12b1e
go: downloading github.com/u-root/u-root v7.0.0+incompatible
go: downloading golang.org/x/net v0.0.0-20210614182718-04defd469f4e
go: downloading golang.org/x/sys v0.0.0-20210630005230-0f9fa26af87c
go: downloading github.com/coreos/go-systemd/v22 v22.3.2
go: downloading github.com/Masterminds/semver v1.5.0
go: downloading github.com/huandu/xstrings v1.3.2
go: downloading github.com/imdario/mergo v0.3.12
go: downloading github.com/mitchellh/copystructure v1.2.0
go: downloading golang.org/x/crypto v0.0.0-20210616213533-5ff15b29337e
go: downloading github.com/u-root/uio v0.0.0-20210528151154-e40b768296a7
go: downloading github.com/mitchellh/reflectwalk v1.0.2
# github.com/DSpeichert/netbootd/dhcpd
dhcpd/arp.go:68:50: undefined: unix.SIOCSARP
```

--- a/.github/workflows/goreleaser.yml
+++ b/.github/workflows/goreleaser.yml
@@ -16,7 +16,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.16.x
+          go-version: 1.17.x
 
       - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v2

--- a/.goreleaser.yml
+++ b/.goreleaser.yml
@@ -9,8 +9,13 @@ builds:
       - -s -w -X github.com/DSpeichert/netbootd/cmd.version={{.Version}} -X github.com/DSpeichert/netbootd/cmd.commit={{.ShortCommit}} -X github.com/DSpeichert/netbootd/cmd.date={{.Date}}
     goos:
       - linux
+      - darwin
     goarch:
       - amd64
+      - arm
+      - arm64
+    goarm:
+      - 7
 
 archives:
   - wrap_in_directory: true

--- a/cmd/arpinject.go
+++ b/cmd/arpinject.go
@@ -1,7 +1,7 @@
 package cmd
 
 import (
-	"github.com/DSpeichert/netbootd/dhcpd"
+	"github.com/DSpeichert/netbootd/dhcpd/arp"
 	"github.com/rs/zerolog/log"
 	"github.com/spf13/cobra"
 	"net"
@@ -31,7 +31,7 @@ var arpInjectCmd = &cobra.Command{
 				Err(err).
 				Msg("cannot parse mac")
 		}
-		if err = dhcpd.InjectArp(parsedIp, parsedMac, dhcpd.ATF_COM, device); err != nil {
+		if err = arp.InjectArp(parsedIp, parsedMac, arp.ATF_COM, device); err != nil {
 			log.Error().
 				Err(err).
 				Msg("cannot inject arp entry")

--- a/dhcpd/arp/arp.go
+++ b/dhcpd/arp/arp.go
@@ -1,0 +1,31 @@
+package arp
+
+import (
+	"syscall"
+)
+
+// ARP Flag values
+// these are not in golang.org/x/sys/unix
+const (
+	// completed entry (ha valid)
+	ATF_COM = 0x02
+	// permanent entry
+	ATF_PERM = 0x04
+	// publish entry
+	ATF_PUBL = 0x08
+	// has requested trailers
+	ATF_USETRAILERS = 0x10
+	// want to use a netmask (only for proxy entries)
+	ATF_NETMASK = 0x20
+	// don't answer this addresses
+	ATF_DONTPUB = 0x40
+)
+
+// https://man7.org/linux/man-pages/man7/arp.7.html
+type arpReq struct {
+	ArpPa   syscall.RawSockaddrInet4
+	ArpHa   syscall.RawSockaddr
+	Flags   int32
+	Netmask syscall.RawSockaddr
+	Dev     [16]byte
+}

--- a/dhcpd/arp/arp_linux.go
+++ b/dhcpd/arp/arp_linux.go
@@ -1,0 +1,49 @@
+//go:build linux && !amd64 && !arm64
+// +build linux,!amd64,!arm64
+
+package arp
+
+import (
+	"net"
+	"os"
+	"syscall"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+// InjectArp injects an ARP entry into dev's ARP table
+// syscalls roughly based on https://www.unix.com/302447674-post3.html
+// see:
+//   https://github.com/torvalds/linux/blob/8cf8821e15cd553339a5b48ee555a0439c2b2742/net/ipv4/arp.c#L1179
+//   https://github.com/torvalds/linux/blob/8cf8821e15cd553339a5b48ee555a0439c2b2742/net/ipv4/arp.c#L1024
+func InjectArp(ip net.IP, mac net.HardwareAddr, flags int32, dev string) (err error) {
+	fd, err := unix.Socket(unix.AF_INET, unix.SOCK_DGRAM, unix.IPPROTO_UDP)
+	if err != nil {
+		return
+	}
+	f := os.NewFile(uintptr(fd), "")
+	defer f.Close()
+
+	return InjectArpFd(uintptr(fd), ip, mac, flags, dev)
+}
+
+func InjectArpFd(fd uintptr, ip net.IP, mac net.HardwareAddr, flags int32, dev string) (err error) {
+	arpReq := arpReq{
+		ArpPa: syscall.RawSockaddrInet4{
+			Family: syscall.AF_INET,
+		},
+		//Flags: 0x02 | 0x04, // ATF_COM | ATF_PERM;
+		Flags: flags,
+	}
+	copy(arpReq.ArpPa.Addr[:], ip.To4())
+	copy(arpReq.ArpHa.Data[:], mac)
+	copy(arpReq.Dev[:], dev)
+
+	_, _, errno := unix.Syscall(unix.SYS_IOCTL, fd, unix.SIOCSARP, uintptr(unsafe.Pointer(&arpReq)))
+	if errno != 0 {
+		return errno
+	}
+
+	return
+}

--- a/dhcpd/arp/arp_linux_64.go
+++ b/dhcpd/arp/arp_linux_64.go
@@ -1,39 +1,19 @@
-package dhcpd
+//go:build linux && (amd64 || arm64)
+// +build linux
+// +build amd64 arm64
+
+package arp
 
 import (
-	"golang.org/x/sys/unix"
 	"net"
 	"os"
 	"syscall"
 	"unsafe"
+
+	"golang.org/x/sys/unix"
 )
 
-// ARP Flag values
-// these are not in golang.org/x/sys/unix
-const (
-	// completed entry (ha valid)
-	ATF_COM = 0x02
-	// permanent entry
-	ATF_PERM = 0x04
-	// publish entry
-	ATF_PUBL = 0x08
-	// has requested trailers
-	ATF_USETRAILERS = 0x10
-	// want to use a netmask (only for proxy entries)
-	ATF_NETMASK = 0x20
-	// don't answer this addresses
-	ATF_DONTPUB = 0x40
-)
-
-// https://man7.org/linux/man-pages/man7/arp.7.html
-type arpReq struct {
-	ArpPa   syscall.RawSockaddrInet4
-	ArpHa   syscall.RawSockaddr
-	Flags   int32
-	Netmask syscall.RawSockaddr
-	Dev     [16]byte
-}
-
+// InjectArp injects an ARP entry into dev's ARP table
 // syscalls roughly based on https://www.unix.com/302447674-post3.html
 // see:
 //   https://github.com/torvalds/linux/blob/8cf8821e15cd553339a5b48ee555a0439c2b2742/net/ipv4/arp.c#L1179

--- a/dhcpd/arp/arp_notlinux.go
+++ b/dhcpd/arp/arp_notlinux.go
@@ -1,0 +1,18 @@
+//go:build !linux
+// +build !linux
+
+package arp
+
+import (
+	"errors"
+	"net"
+)
+
+// InjectArp injects an ARP entry into dev's ARP table
+func InjectArp(ip net.IP, mac net.HardwareAddr, flags int32, dev string) (err error) {
+	return errors.New("not implemented")
+}
+
+func InjectArpFd(fd uintptr, ip net.IP, mac net.HardwareAddr, flags int32, dev string) (err error) {
+	return errors.New("not implemented")
+}

--- a/dhcpd/handler.go
+++ b/dhcpd/handler.go
@@ -5,8 +5,10 @@ package dhcpd
 import (
 	"errors"
 	"net"
+	"runtime"
 	"strings"
 
+	"github.com/DSpeichert/netbootd/dhcpd/arp"
 	mfest "github.com/DSpeichert/netbootd/manifest"
 	"github.com/insomniacslk/dhcp/dhcpv4"
 	"golang.org/x/net/ipv4"
@@ -69,6 +71,7 @@ func (server *Server) HandleMsg4(buf []byte, oob *ipv4.ControlMessage, peer net.
 	if err != nil {
 		server.logger.Error().
 			Err(err).
+			Int("ifIndex", ifIndex).
 			Msg("failed to find local interface")
 		resp = nil
 		goto response
@@ -187,9 +190,9 @@ response:
 			rawConn, err := server.UdpConn.SyscallConn()
 			if device != "" && err == nil {
 				rawConn.Control(func(fd uintptr) {
-					err = InjectArpFd(fd, resp.YourIPAddr, req.ClientHWAddr, ATF_COM, device)
+					err = arp.InjectArpFd(fd, resp.YourIPAddr, req.ClientHWAddr, arp.ATF_COM, device)
 				})
-				if err != nil {
+				if err != nil && runtime.GOOS == "linux" {
 					server.logger.Error().
 						Err(err).
 						Msg("ioctl failed")

